registry/auth/htpasswd/access.go

```diff
@@ -9,13 +9,17 @@ import (
 	"fmt"
 	"net/http"
 	"os"
+	"sync"
 
 	"github.com/docker/distribution/context"
 	"github.com/docker/distribution/registry/auth"
 )
 
 type accessController struct {
 	realm    string
+	path     string
+	modtime  string
+	mtx      *sync.Mutex
```

Comment:
This doesn't need to be a pointer.

`mtx sync.Mutex`

Then it won't need to be initialized when you create the `accessController`.

I suggest calling it `mu` for consistency with the rest of the code.